AnalyticalGraphicsInc/cesium
# Change default Resource.fetchImage flipY to false

In **1.56** we added ImageBitmap support, which cannot be flipped during texture upload, so I exposed an option to `flipY` on fetch instead. Since textures in Cesium are flipped by default, I set the default flipY to true. This now means that anyone who used `fetchImage()` to do anything other than render it in Cesium (such as read pixels, or draw to another canvas), it was now upside down with this release.

This PR changes the `flipY` default on fetchImage to false. **This is still a breaking change compared to 1.55**.  If someone uses `fetchImage` in Cesium, then passes that image to something like a material that uploads it as a texture, it'll now be upside down in **1.56.1** (this PR). If this is a less common case, then I think this breaking change is better than the breaking change in 1.56. 

### Testing this 

The general rule is, if an image is going to be uploaded as a texture, it should be flipped on fetch. If ImageBitmap is NOT supported, then the image will be flipped during upload. If ImageBitmap is supported, then it will be flipped during fetch.

Now that the default flip on fetch is false, we must flip images everywhere when we fetch if we expect to render them in the WebGL context (with the exception of 3D models which don't get flipped on upload). This means:

* All imagery. 
  * I tested this with by cycling through all the imagery layers in the widget picket in the [Hello World](http://localhost:8080/Apps/Sandcastle/index.html) Sandcastle.
  * By confirming all the layers look the same as in 1.55 in the [Imagery Layers](http://localhost:8080/Apps/Sandcastle/index.html?src=Imagery%20Layers%20Manipulation.html) Sandcastle.
* Materials with images
  * Check the "image" dropdown in the [Materials](http://localhost:8080/Apps/Sandcastle/index.html?src=Materials.html) Sandcastle.
  * Check the [Ground Materials](http://localhost:8080/Apps/Sandcastle/index.html?src=development%2FGround%20Primitive%20Materials.html) for good measure.
* CubeMap materials and CubeMaps
  * I couldn't find a good test case for this? See Material.js:880 and loadCubeMap.js
* The PinBuilder 
  * This is subtle, but it's actually upside down in 1.56. See [Map Pins](http://localhost:8080/Apps/Sandcastle/index.html?src=Map%20Pins.html) Sandcastle.
  * Fixing this involved flipping in `TextureAtlas`, which should fix the Billboards too.
* PostProcess effects with image uniforms
  * This one has no Sandcastle. So I [created one](http://localhost:8080/Apps/Sandcastle/index.html#c=jVVhT9swEP0rVr803YJToIAGBQ0VpiGBQCrjy4oqN7mm3hy7cpywgvrfd3acEEqZFrWVY9+9O7+7ey2ZJiWHJ9DklEh4IiPIeZHRB7cXdGP3OlLSMC5Bd0PyMpEEn3yhCpGcS54xA+SYGF3ARK57JxNZImYegwSErLCpe8Wj6nCpcm64knjuw42YNrhicp/OtcouINUAebCzu7dP+0eDweHul5AMBrR/0N8/6h/WUQotEKNLaYSfMcuWAi6YYVGmEhB5VIHfMOlXU1zSVMy63h2k4Wb1mqXRLP4NyeXGtjPjkFOWJIG/vmSZvTUmEFYbzZ2Om6U/ccngtve0T6F55VxtrT1xExlFZLxgCZAcBMQGEu/9xM2CLHi6EPg1tEp/rlmaYXLOQ49VoWOkvILsFpLPlc5I7ljRexckVkLpe/hjCo2lmMgu+fyhLRbDXGUs3TDEqCsuU1JCvEfKqanARkrphEvsg3w7LpoPXrPfgFQcL4nNFfSQofaJ/XGeLnEshw+3dxE06YVbs+idvAPicxLEz9m05jXobQvnQ+6/JosVOG0Rz8in1ptOZ+QzCXZpn+y0jXpo5ZK2Fidbo6Ri+g3LN/JXs/cMWkFDgqBbrrEm2NdAXsj/gLoUtmA0O123rCfWIJ/o5UaVYg+bO61w/POxPaiavyUQdxsG9WC8bUrs8m1d6ifD90f+djgaGqxzIWM7Sq45Ggv7aMCiy7ZkuWvbYoSk3/o5sJNVe61D8vrSdBEGshrC7bpWjem1StVUlaAFW9GlTLutWbWQji/azOkp+fnYCNyCyURsSuo4Rk2T4yWL4bJEQr5XRkFbImnMEMA2sPQYGMFcyWVhzisiGkYyzM3y2jDjlJVbAbud/cKkNtSX2rPGC0UtufM6VRNkZ8TnmsAc5T4J2ni9NzV4f/22LV1qniF4CY8evO7cfyE0tshx+BFt96sl0JvbH+PL6c3tw2Wlm52wM8zNSsBZhfCVZ0uljZXYACtrAIXNKkM0KzBJQ+PccTyMaqdhwkvCk9NJZ+Mfb9IhsWB5jifzQogxf4ZJ52wYof0bN6EYqk96WzWMNVnsnl1Xm5TSYYSv772MUmLGdAvxLw). The Cesium logo should be right side up in full screen.

It's also important to test these in Firefox, where ImageBitmap is not supported. 

* [x] All the tests above work in Firefox.
